core/server/src/main/java/alluxio/worker/block/BlockMetadataManagerView.java

```diff
@@ -229,4 +229,19 @@ public BlockMeta getBlockMeta(long blockId) throws BlockDoesNotExistException {
       return null;
     }
   }
+
+  /**
+   * Return -1 if block does not exist, else return the byte size of the block.
+   *
+   * @param blockId id of the block
+   * @return block size or -1 if the block does not exist
+   * @throws BlockDoesNotExistException if no {@link BlockMeta} for this block id is found
+   */
+  public long getBlockSizeIfExist(long blockId) throws BlockDoesNotExistException {
+    if (mMetadataManager.hasBlockMeta(blockId)) {
+      return mMetadataManager.getBlockMeta(blockId).getBlockSize();
```

Comment:
Just use `mMetadataManager.getBlockMeta(blockId).getBlockSize();` in where you use `getBlockSizeIfExist`, if the block does not exist, `getBlockMeta` will throw `BlockDoesNotExistException`.